node/silkworm/common/log.cpp

```diff
@@ -83,6 +90,19 @@ BufferBase::BufferBase(Level level) : level_(level) {
     }
 }
 
+BufferBase::BufferBase(Level level, std::string_view msg, const std::vector<std::string>& args) : BufferBase(level) {
```

Comment:
`vector<pair<string>>` would probably represent the intent better.

Reply:
Well ... if one side this ensures an even number of items to print it adds ugly braces to inline instantiaton.
Instead of
```
log::Error("Block Validation Error", {"block", std::to_string(block_num_), "err",
                                                      std::string(magic_enum::enum_name<ValidationResult>(res))});
```
we'd have to
```
log::Error("Block Validation Error", {{"block", std::to_string(block_num_)}, {"err",
                                                      std::string(magic_enum::enum_name<ValidationResult>(res))}});
```
Besides there is no strict need to have pairs as we can alwasy write
```
log::Error("Block Validation Error", {"block", std::to_string(block_num_), "err"}) << 
                                                      std::string(magic_enum::enum_name<ValidationResult>(res));
```